tests/rptest/services/kgo_verifier_services.py

```diff
@@ -53,91 +52,236 @@ def __init__(self, context, redpanda, topic, msg_size, custom_node):
+
+    @property
+    def logger(self):
+        return self._parent.logger
+
+    def raise_on_error(self):
+        if self._ex is not None:
+            raise self._ex
+
+    def run(self):
+        try:
+            # Don't assume the HTTP port will be open instantly on startup (although it should be open very soon)
+            self.await_ready()
+
+            self.poll_status()
+        except Exception as ex:
+            self._ex = ex
+            self.logger.exception(
+                f"Error reading status from {self.who_am_i} on {self._node.name}"
+            )
+
+    def is_ready(self):
+        try:
+            r = requests.get(self._parent._remote_url(self._node, "status"))
+        except (requests.exceptions.RequestException,
+                requests.exceptions.ConnectionError) as e:
+            self.logger.debug(
+                f"Status endpoint {self.who_am_i} not ready: {e}")
+            return False
+        except Exception as e:
+            self.logger.debug(
+                f"Status endpoint {self.who_am_i} not ready: {e}")
+            return False
+        else:
+            return r.status_code == 200
+
+    def _ingest_status(self, worker_statuses):
+        reduced = self._status_cls(**worker_statuses[0])
+        for s in worker_statuses[1:]:
+            reduced.merge(self._status_cls(**s))
+
+        self._parent._status = reduced
+
+    def await_ready(self):
+        """
+        Wait for the remote processes http endpoint to come up
+        """
+        if self._ready:
+            return
+
+        wait_until(
+            lambda: self.is_ready() or self._stop_now.is_set(),
+            timeout_sec=5,
+            backoff_sec=0.5,
+            err_msg=
+            f"Timed out waiting for status endpoint {self.who_am_i} to be available"
+        )
+        self._ready = True
+
+    def poll_status(self):
+        while not self._stop_now.is_set():
+            drop_out = self._last_read.is_set()
+
+            r = requests.get(self._parent._remote_url(self._node, "status"),
+                             timeout=5)
+            r.raise_for_status()
+            worker_statuses = r.json()
+            self._ingest_status(worker_statuses)
+
+            if self._status_cls == ProduceStatus:
+                progress = (worker_statuses[0]['sent'] /
+                            float(self._parent._msg_count))
+                self.logger.info(
+                    f"Producer {self.who_am_i} progress: {progress*100:.2f}% {self._parent._status}"
+                )
+
+            if drop_out:
+                # We were asked to clean shutdown and we have done our final
+                # status read
+                return
+            else:
+                self._stop_now.wait(self.INTERVAL)
+
+    def stop_now(self):
+        self._stop_now.set()
+        self.join()
+
+    def clean_shutdown(self):
+        self._last_read.set()
+        self.join()
+
+
+class ValidatorStatus:
```

Comment:
nit: is this meant to be used outside of ConsumerStatus? Maybe consider making it a subclass, just so readers aren't confused why when/how to use this? Alternatively, add a couple lines to describe how this is different from ConsumerStatus.

Reply:
The class structure here mirrors that internally to kgo-verifier, where validator is the common set of state that all consumer worker types have, and the rest of the status is allowed to differ per worker type.

I'll add a docstring.